File: dis-ingest/src/test/java/io/pivotal/dis/ingest/domain/tfl/LineColorTest.java
package io.pivotal.dis.ingest.domain.tfl;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

public class LineColorTest {

    @Test
    public void returnsBackgroundColorForLine() {
        String bakerlooBackgroundColor = LineColor.getBackgroundColorForLine("Bakerloo");
        assertThat(bakerlooBackgroundColor, equalTo("#AE6118"));
    }

    @Test
    public void returnsDefaultBackgroundColorForUnknownLine() {
        String fakeLineBackgroundColor = LineColor.getBackgroundColorForLine("fakerfakefake");
        assertThat(fakeLineBackgroundColor, equalTo("#FFFFFF"));
    }

    @Test
    public void returnsForegroundColorForLine() {
        String bakerlooForegroundColor = LineColor.getForegroundColorForLine("Bakerloo");
        assertThat(bakerlooForegroundColor, equalTo("#FFFFFF"));
    }

    @Test
    public void returnsDefaultForegroundColorForUnknownLine() {
        String fakeLineForegroundColor = LineColor.getForegroundColorForLine("fakerfakefake");
        assertThat(fakeLineForegroundColor, equalTo("#000000"));
    }
}

File: dis-ingest/src/test/kotlin/io/pivotal/dis/ingest/domain/tfl/LineColorTest.kt
package io.pivotal.dis.ingest.domain.tfl

import org.junit.Test

import org.hamcrest.CoreMatchers.equalTo
import org.junit.Assert.assertThat

class LineColorTest {

    @Test
    fun returnsBackgroundColorForLine() {
        val bakerlooBackgroundColor = LineColor.getBackgroundColorForLine("Bakerloo")
        assertThat(bakerlooBackgroundColor, equalTo("#AE6118"))
    }

    @Test
    fun returnsDefaultBackgroundColorForUnknownLine() {
        val fakeLineBackgroundColor = LineColor.getBackgroundColorForLine("fakerfakefake")
        assertThat(fakeLineBackgroundColor, equalTo("#FFFFFF"))
    }

    @Test
    fun returnsForegroundColorForLine() {
        val bakerlooForegroundColor = LineColor.getForegroundColorForLine("Bakerloo")
        assertThat(bakerlooForegroundColor, equalTo("#FFFFFF"))
    }

    @Test
    fun returnsDefaultForegroundColorForUnknownLine() {
        val fakeLineForegroundColor = LineColor.getForegroundColorForLine("fakerfakefake")
        assertThat(fakeLineForegroundColor, equalTo("#000000"))
    }
}
